### ISSUE TITLE
Support headless analysis and reporting in order to integrate CogniCrypt in build pipelines

### ISSUE THREAD

**Issue by Treehopper** [eclipse-cognicrypt/CogniCrypt#226]
I would like to use CogniCrypt as a Maven plug-in that automatically scans my code and reports problems every time a build in Jenkins is running. I assume it should be possible to refactor the core components of the Eclipse plug-in, to make them usable in a Maven plug-in as well. If it is not the scope of this project to provide a Maven plug-in, another project could take care of this, given there are no Eclipse dependencies in those core components.

**Comment by kruegers** [comment]
Hey,

Thanks for the request, but unfortunately it is not entirely clear to me what you would like us to do here. From the title, it seems you want us to extract the analysis into a stand-alone tool. This we already have. It's available [here](https://github.com/CROSSINGTUD/CryptoAnalysis). 

The description reads more like you want us to build a Maven plugin running the analysis. A while ago, we built a [prototype of such a tool](https://github.com/CROSSINGTUD/CryptoAnalysis-MavenPlugin), but haven't really maintained it. We don't know whether it still actually works and currently do not have plans to work on it any further. We would welcome any contribution in that regard, though.

Is either of those what you are looking for?

**Comment by Treehopper** [comment]
Sorry for the confusion. The ideal implementation for me would be a Maven plug-in, but I realized that this may not be the scope of your project. Hence, I tried to phrase it more general in the title to start a discussion that is not restricted to a specific solution. 

CogniCrypt_SAST is pretty much what I was hoping for. I will also have a look at the prototype soon, and see if it can be revived.

In my opinion this can be closed.
Thanks!
